src/cuda.jl

```diff
@@ -1,6 +1,7 @@
 import .CUDA
 
 function Base.cconvert(::Type{MPIPtr}, buf::CUDA.CuArray{T}) where T
+    CUDA.synchronize(CUDA.CuDefaultStream())
```

Comment:
@vchuravy What do you think?

Reply:
Why do we need this again?

cc: @maleadt

Reply:
https://github.com/JuliaParallel/MPI.jl/issues/443